lib/std/math/gdc.zig

```diff
@@ -0,0 +1,26 @@
+/// Find the greatest common divisor (GCD) of two integers.
+const std = @import("std");
+const expect = std.testing.expect;
+
+// Greatest common divisor
+fn gcd(a: anytype, b: anytype) @TypeOf(a) {
+    if (a < 0 or b < 0) @compileError("gcd is only defined for positive numbers (integer)");
```

Comment:
You can't just compileError here; runtime arguments would cause this error to be hit regardless of whether they were positive (because the compiler can't know). Instead, you should just `std.debug.assert(a >= 0 and b >= 0);`.